school_center.py

```diff
@@ -205,10 +198,24 @@ def is_allocated(scode1: str, scode2: str) -> bool:
 remaining = 0       # stores count of non allocated students
 allocations = {}    # to track mutual allocations
 
-OUTPUT_DIR = 'results/'
-create_dir(OUTPUT_DIR)  # Create the output directory if not exists
-with open('{}school-center-distance.tsv'.format(OUTPUT_DIR), 'w', encoding='utf-8') as intermediate_file, \
-        open(OUTPUT_DIR + args.output, 'w', encoding='utf-8') as a_file:
+
+def get_output_dir():
+    dirname = path.dirname(args.output)
+    if(dirname):
+        return dirname
+    else:
+        return DEFAULT_OUTPUT_DIR
+
+def get_output_filename():
+    return path.basename(args.output)
+
+
+output_dirname = get_output_dir()
+output_filename = get_output_filename()
+makedirs(output_dirname, exist_ok=True) # Create the output directory if not exists
+
+with open(path.join(output_dirname, "school-center-distance.tsv"), 'w', encoding='utf-8') as intermediate_file, \
+open(path.join(output_dirname, 'output_filename'), 'w', encoding='utf-8') as a_file:
```

Comment:
should this be just output_filename, not in quotes ?

Reply:
Yes, Thank you, that was my mistake. I corrected it.